apps/sms/js/startup.js

```diff
@@ -112,17 +112,19 @@ var Startup = {
       MessageManager.init();
       Navigation.init();
 
-      // If target panel is different from the default one, let's load remaining
-      // scripts as soon as possible, otherwise we can wait until first page of
-      // threads is loaded and rendered.
-      var panelName = Navigation.getPanelName();
-      if (panelName && panelName !== 'thread-list') {
-        // Initialize default panel only after target panel is ready.
+      var panelName = Navigation.getPanelName() || Navigation.defaultPanel;
+
+      // If initial panel is default one and app isn't run from notification,
+      // then just navigate to it, otherwise we can delay default panel
+      // initialization until we navigate to requested non-default panel.
+      if (panelName === Navigation.defaultPanel &&
+        !navigator.mozHasPendingMessage('notification')) {
+        Navigation.toPanel(panelName);
```

Comment:
Hey @steveck-chung, including you into discussion, since @julienw is on PTO and shouldn't bother with all this :)

What are your thoughts on this?